Source/DafnyCore/AST/BuiltIns.cs

```diff
@@ -152,11 +152,11 @@ public class BuiltIns {
       // declaration of total arrow-type
 
       tps = Util.Map(Enumerable.Range(0, arity + 1), x => x < arity ?
-        new TypeParameter(tok, "T" + x, TypeParameter.TPVarianceSyntax.Contravariance) :
-        new TypeParameter(tok, "R", TypeParameter.TPVarianceSyntax.Covariant_Strict));
+        new TypeParameter(RangeToken.NoToken, new Name("T" + x), TypeParameter.TPVarianceSyntax.Contravariance) :
+        new TypeParameter(RangeToken.NoToken, new Name("R"), TypeParameter.TPVarianceSyntax.Covariant_Strict));
       tys = tps.ConvertAll(tp => (Type)(new UserDefinedType(tp)));
       id = new BoundVar(tok, "f", new UserDefinedType(tok, partialArrow.Name, partialArrow, tys));
```

Comment:
Could you replace `tok` here as well?

Reply:
I can't yet, because many of the AST nodes still take an IToken and not a RangeToken as a first argument.